lib/resources/resourceBase.js

```diff
@@ -64,7 +64,10 @@ var ResourceBase = function (endpoint, config) {
         }
 
         if (body && body.error) {
-          return reject(body.error);
+          var error = new Error(body.error.message + ' (' + body.error.status_code + ')');
+          error.statusCode = body.error.status_code;
+          error.statusMessage = body.error.message;
```

Comment:
That sounds good. Why snake_case instead of camelCase for the status code though?